src/api/api.tsx

```diff
@@ -5,68 +5,154 @@ import {
   NotFound,
   PermissionDenied,
 } from '../errors/errors';
-
-const LOCALSTORAGE_TOKEN_NAME = 'zaaktypenbeheer_token';
-localStorage.setItem(LOCALSTORAGE_TOKEN_NAME, import.meta.env.VITE_API_LOGIN_TOKEN);
-
-interface APIResponse {
-  detail: string;
-  [key: string]: any;
-}
+import { CSRFToken } from './headers';
 
 class Api {
-  api_token: string | null;
   api_url: string;
+  fetchDefaults: any;
 
   constructor() {
-    this.api_token = localStorage.getItem(LOCALSTORAGE_TOKEN_NAME);
-    this.api_url = import.meta.env.VITE_BASE_API_URL;
+    this.api_url = import.meta.env.VITE_BASE_API_URL as string;
+    this.fetchDefaults = {
+      credentials: 'include', // required for Firefox 60, which is used in werkplekken
+    };
+  }
+
+  getEndpoint(url: string) {
+    if (!this.api_url.endsWith('/')) this.api_url += '/';
+    if (url.startsWith('/')) url = url.substring(1);
+    if (!url.endsWith('/') && url !== '') url += '/';
+    return this.api_url + url;
   }
 
-  loggedIn = () => {
-    return Boolean(this.api_token);
-  };
+  async getUser() {
+    try {
+      return await this.get('users/me/');
+    } catch (_err) {
+      return null;
+    }
+  }
 
-  signIn = async (formData: FormData) => {
-    const data = {
-      username: formData.get('username'),
-      password: formData.get('password'),
-    };
+  updateStoredHeadersValues(headers: Headers) {
+    const CSRFTokenValue = headers.get(CSRFToken.headerName);
+    if (CSRFTokenValue) {
+      CSRFToken.setValue(CSRFTokenValue);
+    }
+  }
 
-    const response = await this.post('api-token-auth', data);
+  async signIn(data: any) {
+    await this.get(''); // get CSRF token
+    await this.post('auth/login/', data);
+    return await this.getUser();
+  }
+
+  async signOut() {
+    await this.post('auth/logout/', {});
+  }
```

Comment:
I think it might be good to separate the concerns here. In the api file have the utilities to make api calls and in the Auth have the ones needed to deal with signIn/signOut.

Reply:
Ok! Will change this to `Auth`